tools/build-tool/data/android/integrated/app/src/main/java/net/sourceforge/castleengine/ServiceMiscellaneous.java

```diff
@@ -91,6 +105,10 @@ public boolean messageReceived(String[] parts)
             Toast.makeText(getActivity().getApplicationContext(), parts[1], Toast.LENGTH_SHORT).show();
             return true;
         } else
+        if (parts.length == 2 && parts[0].equals("change-keyboard-state")) {
+            changeKeyboardState(parts[1]);
+            return true;
```

Comment:
Use `stringToBoolean` to convert `parts[1]` to boolean, and pass it as a boolean to `changeKeyboardState`. So 

```
changeKeyboardState(stringToBoolean(parts[1]));
```

This is consistent with our other services, e.g. https://github.com/castle-engine/castle-engine/blob/master/tools/build-tool/data/android/integrated-services/google_play_games/app/src/main/java/net/sourceforge/castleengine/ServiceGooglePlayGames.java#L631